src/update-feeds-loops/submit-transactions.ts

```diff
@@ -199,37 +244,38 @@ export const submitTransactions = async (
   api3ServerV1: Api3ServerV1,
   updatableDataFeeds: UpdatableDataFeed[],
   blockNumber: number
-) =>
-  Promise.all(
+) => {
+  const {
+    config: { walletDerivationScheme },
+  } = getState();
+  if (walletDerivationScheme.type === 'fixed') {
+    return updatableDataFeeds.length > 0
+      ? Array.from({ length: updatableDataFeeds.length }).fill(
```

Comment:
I only wasn't satisfied with this part looks too hacky for me. If one doesn't know `.fill` implementation like me can think that `.fill` calls its input for length of array that is going to be filled. Additionally, with this addition, typescript couldn't resolve the output type of main function `submitTransactions` and just says `unknown`. 

I checked that we are only using output of this function to calculate success count [here](https://github.com/api3dao/airseeker-v2/blob/d17db3c82c37c612a5971276394fb35bf6502e02/src/update-feeds-loops/update-feeds-loops.ts#L313). So my alternative solution would be implementing this function similar to following. Wdyt?

```typescript
export const submitTransactions = async (
  chainId: string,
  providerName: string,
  provider: ethers.JsonRpcProvider,
  api3ServerV1: Api3ServerV1,
  updatableDataFeeds: UpdatableDataFeed[],
  blockNumber: number
) => {
  if (updatableDataFeeds.length === 0) return 0;
  const {
    config: { walletDerivationScheme },
  } = getState();

  if (walletDerivationScheme.type === 'fixed') {
    const result = await submitBatchTransaction(
      chainId,
      providerName,
      provider,
      api3ServerV1,
      updatableDataFeeds,
      blockNumber
    );
    return result ? updatableDataFeeds.length : 0;
  }

  const result = await Promise.all(
    updatableDataFeeds.map(async (dataFeed) =>
      submitTransaction(chainId, providerName, provider, api3ServerV1, dataFeed, blockNumber)
    )
  );
  return result.filter(Boolean).length;
};
```

Reply:
Looks really neat 🔥 Thanks for the taking the time to actually write the code. I'll apply that change and fix the tests 🙏🏻